### PR TITLE
tests: Add missing test in String

### PR DESCRIPTION
**Pull request by saulecabrera**
Adds missing tests for the following:

- `codePointAt`
- `concat`
- `fromCharCodes`

@dcodeIO I also found out that the documentation lists `String#fromCodePoints` as a static member, but there's no actual definition for it. Should this be defined or removed from the docs instead? If it's a documentation thing, I can submit a PR to fix it.

- [x] I've read the contributing guidelines

### CODE REVIEW

**Comment by SebastianSpeitel** [comment]
If we are already at `fromCharCodes`, maybe we could look further into the signature to allow arrays of different element sizes.
I already prepared 2 options:
https://github.com/SebastianSpeitel/assemblyscript/commit/fcf2fd9e88723d51daae696d75ba89922a3955b1
https://github.com/SebastianSpeitel/assemblyscript/commit/08a4f4dbd87636c108848822b6316bac59281f3a

**Comment by saulecabrera** [comment]
> If we are already at `fromCharCodes`, maybe we could look further into the signature to allow arrays of different element sizes.
> I already prepared 2 options:
> [SebastianSpeitel@fcf2fd9](https://github.com/SebastianSpeitel/assemblyscript/commit/fcf2fd9e88723d51daae696d75ba89922a3955b1)
> [SebastianSpeitel@08a4f4d](https://github.com/SebastianSpeitel/assemblyscript/commit/08a4f4dbd87636c108848822b6316bac59281f3a)

I'm probably missing something here, but could you elaborate on what would be the advantage of doing this? In https://github.com/SebastianSpeitel/assemblyscript/commit/08a4f4dbd87636c108848822b6316bac59281f3a you're still checking if the value of the units is an integer, so is it just for the semantics of the function?

**Comment by SebastianSpeitel** [comment]
It allows you to use all the different integer types not just i32, where the first 2 bytes are discarded anyways.

File: tests/compiler/std/string.ts [hunk 1]
```diff
@@ -770,6 +775,12 @@ assert(dtoa(1.1e+128) == "1.1e+128");
 assert(dtoa(1.1e-64) == "1.1e-64");
 assert(dtoa(0.000035689) == "0.000035689");
 
+
+// concat
+
+assert("Hello ".concat("World") == "Hello World");
+assert("".concat("bar") == "bar");
```
Comment (MaxGraey) [inline tests/compiler/std/string.ts:782]: Could you add more cases?
```ts
"bar".concat("") == "bar"
"".concat("") == ""
```

Comment (saulecabrera) [reply tests/compiler/std/string.ts:782]: Done.

BTW, is this intended https://github.com/AssemblyScript/assemblyscript/blob/master/std/assembly/string.ts#L87? 

Why is the null check necessary? Isn't the function signature enough?

Comment (MaxGraey) [reply tests/compiler/std/string.ts:782]: See this patch: https://github.com/AssemblyScript/assemblyscript/pull/1682
null checks was necessary in the past. But for now it's not necessary

**Review by MaxGraey — APPROVED** [review]

**Comment by dcodeIO** [comment]
Thank you! :)

**Pull request merged**